pkg/asset/ignition/bootstrap.go

```diff
@@ -0,0 +1,335 @@
+
+	enabled := true
+	disabled := false
+	bootkubeService := ignition.NewSystemdUnit("bootkube.service", templates.BootkubeSystemdContents)
+	bootkubeService.Enabled = &disabled
+	tectonicService := ignition.NewSystemdUnit("tectonic.service", templates.TectonicSystemdContents)
+	tectonicService.Enabled = &enabled
+	alwaysUpdateCACertsService := ignition.NewSystemdDropin("update-ca-certificates.service", "10-always-update-ca-certificates.conf", templates.AlwaysUpdateCACertsSystemdContents)
+	alwaysUpdateCACertsService.Enabled = &enabled
+	kubeletService := ignition.NewSystemdUnit("kubelet.service", applyTemplateData(templates.BootstrapKubeletSystemdContents, templateData))
+	kubeletService.Enabled = &enabled
+	ignitionConfig.Systemd.Units = append(
+		ignitionConfig.Systemd.Units,
+		bootkubeService,
+		tectonicService,
+		alwaysUpdateCACertsService,
+		kubeletService,
+	)
+
+	ignitionConfig.Passwd.Users = append(
+		ignitionConfig.Passwd.Users,
+		ignition.NewUser(installConfig.Admin.Email, installConfig.Admin.SSHKey),
+	)
+
+	data, err := ignition.WriteConfig(ignitionConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &asset.State{
+		Contents: []asset.Content{
+			{
+				Name: filepath.Join(a.directory, "bootstrap.ign"),
+				Data: data,
+			},
+		},
+	}, nil
+}
+
+// getTemplateData returns the data to use to execute bootstrap templates.
+func (a *bootstrap) getTemplateData(installConfig *types.InstallConfig) (*bootstrapTemplateData, error) {
+	clusterDNSIP, err := getClusterDNSIP(installConfig)
+	if err != nil {
+		return nil, err
+	}
+	etcdEndpoints := make([]string, masterCount(installConfig))
+	for i := range etcdEndpoints {
+		etcdEndpoints[i] = fmt.Sprintf("https://%v:2379", etcdName(installConfig, i))
+	}
+	return &bootstrapTemplateData{
+		KubeletImageURL:     "openshift/origin-node",
+		KubeletImageTag:     "latest",
+		ClusterDNSIP:        clusterDNSIP,
+		CloudProvider:       getCloudProvider(installConfig),
+		CloudProviderConfig: getCloudProviderConfig(installConfig),
+		DebugConfig:         "",
+		NodeTaintsParam:     "--register-with-taints=node-role.kubernetes.io/bootstrap=:NoSchedule",
+		KubeCoreRenderImage: "quay.io/coreos/kube-core-renderer-dev:436b1b4395ae54d866edc88864c9b01797cebac1",
+		TNCOperatorImage:    "quay.io/coreos/tectonic-node-controller-operator-dev:436b1b4395ae54d866edc88864c9b01797cebac1",
+		EtcdCertSignerImage: "quay.io/coreos/kube-etcd-signer-server:678cc8e6841e2121ebfdb6e2db568fce290b67d6",
+		EtcdctlImage:        "quay.io/coreos/etcd:v3.2.14",
+		BootkubeImage:       "quay.io/coreos/bootkube:v0.10.0",
+		HyperkubeImage:      "openshift/origin-node:latest",
+		EtcdCluster:         strings.Join(etcdEndpoints, ","),
+	}, nil
+}
+
+func (a *bootstrap) addBootstrapConfigFilesToIgnitionConfig(ignitionConfig *ignitionTypes.Config, dependencies map[asset.Asset]*asset.State) {
+	// TODO (staebler) - missing the following from assets step
+	//     /opt/tectonic/manifests/cluster-config.yaml
```

Comment:
Do we want to address these in this PR?  If not, I'd recommend dropping `addBootstrapConfigFilesToIgnitionConfig` completely and coming back around to these in follow-up work if/when we decide we need them.